cmd/create/operatorroles/cmd.go

```diff
@@ -149,12 +149,21 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
-	if cluster.State() != cmv1.ClusterStatePending {
-		reporter.Errorf("Cluster '%s' is %s and does not need additional configuration.",
-			clusterKey, cluster.State())
+	// Check to see if IAM operator roles have already created
+	missingRoles, err := validateOperatorRoles(awsClient, cluster)
+	if err != nil {
+		reporter.Errorf("Unable to find validate operator roles exist: %s", err)
 		os.Exit(1)
 	}
 
+	if len(missingRoles) == 0 {
+		if cluster.State() != cmv1.ClusterStatePending {
+			reporter.Errorf("Cluster '%s' is %s and does not need additional configuration.",
+				clusterKey, cluster.State())
+			os.Exit(1)
+		}
+	}
+
 	if cluster.AWS().STS().RoleARN() == "" {
```

Comment:
This check becomes redundant this late in the flow. I suggest moving it above the `validateOperatorRoles` call just as an extra safety measure.

Reply:
(same with the one in the `oidcprovider/cmd.go` file, but the GitHub interface doesn't let me add a comment there)